### PR TITLE
[Google Connectors family]: Restructure

### LINKED ISSUE #601: [Google Connectors family]: Restructure
Refine the structure of connectors and introduce a very basic Google base project where common code will be stored.

### PR DESCRIPTION
**Pull request by mrebrii**
## Description

Refine the structure of connectors and introduce a very basic Google base project , which has all common connectors code .

## Related issues

closes #601

### CODE REVIEW

File: bundle/pom.xml [hunk 1]
```diff
@@ -96,7 +96,7 @@
       </dependency>
       <dependency>
         <groupId>io.camunda.connector</groupId>
-        <artifactId>connector-aws-lambda</artifactId>
```
Comment (igpetrov) [inline bundle/pom.xml:99]: Why did you remove lambda?

File: connectors/google/google-base/src/main/java/io/camunda/google/model/Authentication.java [hunk 2]
```diff
@@ -1,10 +1,10 @@
 /*
  * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
- * under one or more contributor license agreements. Licensed under a proprietary license.
- * See the License.txt file for more information. You may not use this file
- * except in compliance with the proprietary license.
+ *             under one or more contributor license agreements. Licensed under a proprietary license.
+ *             See the License.txt file for more information. You may not use this file
+ *             except in compliance with the proprietary license.
```
Comment (igpetrov) [inline connectors/google/google-base/src/main/java/io/camunda/google/model/Authentication.java:5]: There's something wrong with your indentation. It's likely in related `pom.xml`

**Review by igpetrov — CHANGES_REQUESTED** [review]
LGTM. Minor suggestions.

Could you please also make a regression test for google drive connector to make sure it's still working. If you don't have creds - please reach out to me, I will share with you.

Thank you! Great work!

File: connectors/google/google-base/src/main/java/io/camunda/google/model/AuthenticationType.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ *             except in compliance with the proprietary license.
+ */
+package io.camunda.google.model;
+
+import com.google.api.client.util.Value;
+import com.google.gson.annotations.SerializedName;
+
+public enum AuthenticationType {
+  @Value("bearer")
+  @SerializedName("bearer")
```
[the remark this answers is not in the document]
Comment (mrebrii) [reply connectors/google/google-base/src/main/java/io/camunda/google/model/AuthenticationType.java:14]: I can't find a reason, but when using custom Gson factory instead of default, the parser can't parse the value without ```@SerializedName```. It just came to me now that I haven't tried without ```@Value``` annotation. I'll try it and based on the result remove it or close the conversation

[the remark this answers is not in the document]
Comment (mrebrii) [reply connectors/google/google-base/src/main/java/io/camunda/google/model/AuthenticationType.java:14]: Without ```@Value``` default parser doesn't work. So, leave both

File: connectors/google/google-base/src/main/java/io/camunda/google/model/Authentication.java (hunk 2 above)
Comment (mrebrii) [reply connectors/google/google-base/src/main/java/io/camunda/google/model/Authentication.java]: fixed

File: connectors/google/google-base/src/main/java/io/camunda/google/model/AuthenticationType.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (igpetrov) [reply connectors/google/google-base/src/main/java/io/camunda/google/model/AuthenticationType.java]: Please take a look at other enums and try to find a root cause. I believe, we use only one of those.

[the remark this answers is not in the document]
Comment (mrebrii) [reply connectors/google/google-base/src/main/java/io/camunda/google/model/AuthenticationType.java]: Created separate issue #606

**Review by igpetrov — APPROVED** [review]
